float8_experimental/float8_utils.py

```diff
@@ -75,7 +75,13 @@ def tensor_to_amax(x, distributed_reduction=False):
     # If the user did not ask for it, assume that it will
     # happen elsewhere.
     if distributed_reduction and dist.is_initialized():
-        dist.all_reduce(amax, op=dist.ReduceOp.MAX)
+        # TODO: Dynamo rewriting synchronous in-place collectives fails:
+        # https://github.com/pytorch/pytorch/issues/120082
+        # Use functional all-reduce to avoid graph breaking.
+        amax = dist._functional_collectives.all_reduce(
+            amax, "MAX", list(range(dist.get_world_size()))
```

Comment:
Ranks + tag as the process group identifier has been deprecated. Can we pass `dist.group.WORLD` or `dist.group.WORLD.group_name` here?

Reply:
If we are not landing this PR, then is it okay to leave the call as `dist.all_reduce(amax, op=dist.ReduceOp.MAX)` and wait for your Dynamo rewrite changes?

Reply:
Oh the changes you are referring to are for rewriting functional collective. The options I mentioned above should already work :)

Let me know if it doesn't though.